packages/salesforcedx-vscode-core/src/util/authInfo.ts

```diff
@@ -78,12 +78,11 @@ export class OrgAuthInfo {
     }
   }
 
-  public static async getUsername(usernameOrAlias: string): Promise<string> {
+  public static async getUsername(
+    usernameOrAlias: string
+  ): Promise<string | undefined> {
```

Comment:
This became an issue after I added the retry in `workspaceOrgType.ts`. Turns out if `username` was `undefined` we ended up returning the original parameter provided, making it difficult to identify if we had the correct info to keep going. This change simplifies the method and is the reason why I had to update all the other code outside of `workspaceOrgType.ts`.